src/mgmt/srv_drpc.c

```diff
@@ -281,17 +281,33 @@ conv_req_props(daos_prop_t **out_prop, bool set_props,
 }
 
 static int
-create_pool_props(daos_prop_t **out_prop, char *owner, char *owner_grp,
+create_pool_props(daos_prop_t **out_prop, uint32_t numsvcreps, char *owner, char *owner_grp,
 		  const char **ace_list, size_t ace_nr)
 {
+	uint64_t	out_svc_rf = DAOS_PROP_PO_SVC_REDUN_FAC_DEFAULT;
 	char		*out_owner = NULL;
 	char		*out_owner_grp = NULL;
 	struct daos_acl	*out_acl = NULL;
 	daos_prop_t	*new_prop = NULL;
-	uint32_t	entries = 0;
+	uint32_t	entries = 1;
 	uint32_t	idx = 0;
 	int		rc = 0;
 
+	if (numsvcreps > 0) {
+#ifndef DRPC_TEST
+		out_svc_rf = ds_pool_svc_rf_from_nreplicas(numsvcreps);
+#else
+		if (numsvcreps % 2 == 0)
```

Comment:
Shall we return error rather than automatically change this to make raft work?

Reply:
I think we don't report an error when users specify even numbers to "dmg pool create -v <number>" currently. My intention is to gradually retire the "-v" option and move to "-P svc_rf:...". Hence, an error is probably not necessary. But I'm open to changes here.

Reply:
I started a PR ages ago (#7409) to disallow non-odd svc reps on pool creation, but it turned into an ordeal because of all the tests that were broken by it, and I lost steam on trying to finish it up. I think I prefer the approach in this PR, to be honest. If an admin specifies an even number of replicas, it is likely due to either a typo or a misunderstanding of how raft works, and in either case I think it's fine to just fix it for them.

@liw: Given that you are adding the new property as part of this PR, do you think it's worth just pushing forward and converting the `-v` option over to that property? I could push a small addition to this PR that handles that on the dmg side if you'd like.

Reply:
Thanks, @mjmac. I agree in principle, and actually did the dmg changes to mark "-v" as deprecated initially. The difficulty is that to complete the conversion I'd have to come up with a way to change ftest... Probably better to leave that out of this PR, as I'm starting to "losing steam" on this one. :)